src/main/java/cz/smarteon/loxone/user/UserGroup.java

```diff
@@ -0,0 +1,123 @@
+package cz.smarteon.loxone.user;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import cz.smarteon.loxone.LoxoneUuid;
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+import lombok.Setter;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@Getter
+public class UserGroup {
+
+    /**
+     * Enum class containing constants for usergroup type
+     */
+    @RequiredArgsConstructor
+    public enum UserGroupType {
+        NORMAL(0),
+        @Deprecated
+        DEPRECATED_ADMIN(1),
+        ALL(2),
+        NONE(3),
+        ADMIN(4);
+
+        @Getter
+        private final int value;
+    }
+
+    /**
+     * Enum class containing constants for usergroup rights
+     */
+    @RequiredArgsConstructor
+    public enum UserGroupRights {
+        NONE(0),
+        WEB(1),
+        CONFIG(4),
+        FTP(8),
+        TELNET(10),
+        OPERATING_MODES(20),
+        AUTOPILOT(40),
+        EXPERT(80),
+        USER_MANAGEMENT(100),
+        ADMIN(4294967295L);
+
+        @Getter
+        private final long value;
+
+        public static UserGroupRights getValue(long value) {
+            for(UserGroupRights e: UserGroupRights.values()) {
+                if(e.value == value) {
+                    return e;
+                }
+            }
+            return null;
+        }
+    }
+
+    /**
+     * UUID of this usergroup, should be unique
+     */
+    private final @NotNull LoxoneUuid uuid;
+
+    /**
+     * Name of this usergroup, should be unique
+     */
+    private final @Nullable String name;
+
+    /**
+     * Description of this usergroup
+     */
+    @Setter
+    private @Nullable String description;
+
+    /**
+     * Type of this usergroup
+     * <pre>
+     * 0 = Normal
+     * 1 = Admin (deprecated)
+     * 2 = All
+     * 3 = None
+     * 4 = AllAccess - Admin
+     */
+    @Setter
+    private int type;
+
+    /**
+     * Rights of this usergroup
+     */
+    @JsonDeserialize(using = UserGroupRightsDeserializer.class)
+    @Setter
+    private long userRights;
+
+    public UserGroup(@NotNull LoxoneUuid uuid) {
+        this.uuid = uuid;
+        this.name = null;
+        this.description = null;
+    }
+
+    public UserGroup(@NotNull LoxoneUuid uuid, @NotNull String name) {
+        this.uuid = uuid;
+        this.name = name;
+        this.description = null;
+    }
+
+    @JsonCreator
+    public UserGroup(
+            @JsonProperty(value = "uuid", required = true) @NotNull LoxoneUuid uuid,
+            @JsonProperty(value = "name") @Nullable String name,
+            @JsonProperty(value = "description") @Nullable String description,
+            @JsonProperty(value = "type") @Nullable UserGroupType type,
```

Comment:
Depending on enums in deserialization can be dangeorous - in case Loxone will add new user group type in the future the library will be broken. Take a look for instance how it's solved with `MiniserverType` - using UNKNOWN and `@JsonCreator` method - no need for extra deserializer etc.

Reply:
hmm and if it's used for deserialization then also the backing field could be of enumerated type... this way it's kind of upside down